libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/KeyValueEncryptedFileStore.java

```diff
@@ -344,21 +338,34 @@ private boolean isKeyValid(String key, String operation) {
         return files == null ? new File[0] : files;
     }
 
-    InputStream getStream(File file, String encryptionKey) throws IOException, GeneralSecurityException {
-        FileInputStream f = new FileInputStream(file);
-        DecrypterInputStream inputStream = new DecrypterInputStream(f, encryptionKey);
-        return inputStream;
+    InputStream getStream(File file, String encryptionKey) throws IOException {
```

Comment:
You won't be able to read back values written with kv-store from 8.2 because Encryptor.encrypt/decrypt does base 64 encoding/decoding (which is a waste if you don't need a string back). See [encryptBytes](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java#L290) and [decryptBytes](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java#L200). That's the reasons why the tests [testWriteWithEncryptorAndReadThroughStream and testWriteThroughStreamAndReadWithEncryptor](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/test/SalesforceAnalyticsTest/src/com/salesforce/androidsdk/analytics/security/EncrypterStreamTest.java#L84) are commented out.
Maybe we could have encrypt and decrypt methods that take bytes / return bytes and don't base 64 encode or decode.

Reply:
Right - I deliberately left out the upgrade case, I don't think this is being used currently. I will look into adding the vanilla `encryptBytes` and `decryptBytes` methods though.

Reply:
Using new method now.